tests/testlib/Makefile

```diff
@@ -27,4 +29,4 @@ libtests2n.so: ${OBJS}
 	${CC} ${CFLAGS} -shared ${LDFLAGS} -o libtests2n.so ${OBJS}
 
 libtests2n.dylib: ${OBJS}
-	test ! -f /usr/lib/libSystem.dylib || libtool ${LDFLAGS} -dynamic -o libtests2n.dylib ${OBJS}
+	${CC} ${ARCHFLAGS} ${LDFLAGS} -dynamiclib -o libtests2n.dylib ${OBJS}
```

Comment:
I think this will break on non OS-X platforms? the test was checking (primitively) to see if this is an OS X host and if we should try to build a dylib.